### PR TITLE
#1253 Disable password-protected messages when FES is not used

### LINKED ISSUE #1253: disable password-protected messages if FES is not used

### PR DESCRIPTION
**Pull request by sosnovsky**
This PR enables password-protected messages only when FES is used

close #1253

----------------------------------

**Tests** _(delete all except exactly one)_:
- Tests added or updated

--------------------------------

### To be filled by reviewers

I have reviewed that this PR... _(tick whichever items you personally focused on during this review)_:
- [ ] addresses the issue it closes (if any)
- [ ] code is readable and understandable
- [ ] is accompanied with tests, or tests are not needed
- [ ] is free of vulnerabilities
- [ ] is documented clearly and usefully, or doesn't need documentation

### CODE REVIEW

**Comment by sosnovsky** [comment]
I think that after finishing and merging https://github.com/FlowCrypt/flowcrypt-ios/pull/1331 we should also update `EnterpriseServerApi` to have `email` property (which will be passed from `AppContextWithUser`), so we don't need to pass it manually for every method.
Then we'll be able to fetch `fesUrl` only once and store result locally, while currently it's fetched for each request.

**Comment by sosnovsky** [comment]
@tomholub I have 2 remaining tasks - https://github.com/FlowCrypt/flowcrypt-ios/issues/1327 from `0.8.0` milestone and https://github.com/FlowCrypt/flowcrypt-ios/issues/1120 from `0.9.0`
Should I also take some other tasks from `0.8.0` like https://github.com/FlowCrypt/flowcrypt-ios/issues/953 or https://github.com/FlowCrypt/flowcrypt-ios/issues/954 ?

File: FlowCrypt/Functionality/Services/Client Configuration Service/ClientConfigurationService.swift [hunk 1]
```diff
@@ -34,7 +34,8 @@ extension ClientConfigurationService: ClientConfigurationServiceType {
     func fetch(for user: User) async throws -> ClientConfiguration {
         do {
             let raw = try await server.getClientConfiguration(for: user.email)
-            try local.save(for: user, raw: raw)
+            let fesUrl = try await server.getActiveFesUrl(for: user.email)
+            try local.save(for: user, raw: raw, fesUrl: fesUrl)
             return ClientConfiguration(raw: raw)
```
Comment (tomholub) [inline FlowCrypt/Functionality/Services/Client Configuration Service/ClientConfigurationService.swift:39]: from my memory, `getClientConfiguration` also calls `getActiveFesUrl` separately, so we are calling twice. Can we optimize that? 

`getActiveFesUrl` is to figure out if FES is running there or not (once discovered at least once as running, we should save that and always expect it going forward)

`getClientConfiguration` is to fetch the actual config, but it should have to call `getActiveFesUrl` each time if we already saved it as existing at least once.

Comment (sosnovsky) [reply FlowCrypt/Functionality/Services/Client Configuration Service/ClientConfigurationService.swift:39]: Yes, it'll be fixed by https://github.com/FlowCrypt/flowcrypt-ios/issues/1348 - `fesUrl` will be fetched only once and then reused.

Comment (tomholub) [reply FlowCrypt/Functionality/Services/Client Configuration Service/ClientConfigurationService.swift:39]: ok

**Review by tomholub — APPROVED** [review]

**Pull request merged**